client/my-sites/plugins/access-control.js

```diff
@@ -6,32 +6,23 @@ import i18n from 'i18n-calypso';
 /**
  * Internal Dependencies
  */
-import sitesList from 'lib/sites-list';
 import config from 'config';
 import notices from 'notices';
 
-const sites = sitesList();
-
-const hasErrorCondition = ( site, type ) => {
-	const errorConditions = {
-		notMinimumJetpackVersion: site && ! site.hasMinimumJetpackVersion && site.jetpack,
-		notRightsToManagePlugins: sites.initialized && ! sites.canManageSelectedOrAll()
-	};
-	return errorConditions[ type ];
+const getHasRightsToManagePlutins = ( selectedOrAll ) => {
+	return selectedOrAll.some( ( site ) => site.capabilities && site.capabilities.manage_options );
 };
 
-const hasRestrictedAccess = ( site ) => {
-	site = site || sites.getSelectedSite();
-
+const hasRestrictedAccess = ( site, hasRightsToManagePlutins, isMinJetpackVersionValidationFailed ) => {
```

Comment:
Similarly I think the notice behavior can be controlled by the component directly, or as a side effect of a related dispatched action: https://github.com/Automattic/wp-calypso/blob/master/client/state/notices/middleware.js